### DISCUSSION TITLE
Are there any methods to run the object transition manually ?

### DISCUSSION THREAD

**Discussion by kietheros** [minio/minio#18583] [Q&A] [answered]
I set up some rules to transition data to a remote tier. **I wonder how the scanner marks a file matching a rule ?** <br>
Probably, it's set in the metadata of the file. Minio doesn't have an embedded database with indexing, so the scanner always scans all objects to know what to do with them. This work isn't really optimized when dealing with a huge number of files.

I save 800k small files in real-time every day, and these files should be moved to the remote tier after one day. I have set the scanner speed to fast. I checked some files in a folder that was created three days ago, and there still exist some files that have been not moved to the remote. The scanner's file transition speed must be as fast as the file creation speed, otherwise my 2TB SSD storage will be full after a week.

When saving files, I have partitioned the data day-by-day:
- data/files/2023/12/01/...
- data/files/2023/12/02/...

I think the scanner doesn't need to scan every object to check with transition rules. Are there any methods to run the transition manually for a prefix ? I will set up that command in the crontab ?

**Comment by harshavardhana** [comment, marked as answer]
Yeah if you just do mc ls -r it triggers transition rules

**Reply by kietheros** [reply]
Is my understanding of the Minio scanner correct? It always scans all objects because Minio doesn't have a database for indexing object created_time. Using `mc ls -r data/files/2023/12/01/ ` is similar to scanner scanning.

**Reply by kietheros** [reply]
@harshavardhana Sorry for bother you, I cannot ask minio to run transition rules manually, `mc ls -r  myminio/data/files/2023/12/01/` does not trigger transition rules as you said.

**Comment by klauspost** [comment]
> these files should be moved to the remote tier after one day.

Tiering does not *move* your objects. It merely puts the object data on remotes - but keeps metadata. So your objects should still be accessed through MinIO and will remain visible there.

**Reply by kietheros** [reply]
Of course, I understand that. I use this SSD minio as a caching layer for read/writing data in current day. Files need to be transitioned to remote tier after one day. I remove files after one month, so 2TB is enough for saving all metadata of files.

**Comment by kietheros** [comment]
@harshavardhana 
I intend to set the scanner speed to the slowest. I write a background job to manage the object life cycle manually:
### Expiring Objects: delete files by prefix, for example:

```bash
mc rm -r /data/files/2023/11/02/
This task will run at 23:00 every day and stop at 05:00 the next day.
```
### Transitioning Objects: I will use the command:

```bash
mc ls -r /data/files/2023/12/02/
This task will also run at 23:00 every day and stop at 05:00 the next day.
```
Because I know the exact files that need to be removed or transitioned, my job will not need to scan all objects. Could you give me some comments about my solution ?
